assets/src/edit-story/components/panels/index.js

```diff
@@ -75,68 +76,69 @@ function intersect(a, b) {
+      { type: NO_SELECTION, Panel: NoSelectionPanel },
+    ];
   }
 
   const isBackground = elements.length === 1 && elements[0].isBackground;
 
-  // Panels to always display, independent of the selected element.
-  const sharedPanels = [{ type: ACTIONS, Panel: ActionsPanel }];
-
-  let selectionPanels = [];
   // Only display background panel in case of background element.
   if (isBackground) {
-    selectionPanels = [{ type: BACKGROUND, Panel: BackgroundPanel }];
+    const panels = [
+      { type: PAGE, Panel: PageBackgroundPanel },
+      { type: BACKGROUND, Panel: BackgroundPanel },
+    ];
     // If the selected element's type is video, display poster panel, too.
     if ('video' === elements[0].type) {
-      selectionPanels.push({ type: VIDEO_POSTER, Panel: VideoPosterPanel });
+      panels.push({ type: VIDEO_POSTER, Panel: VideoPosterPanel });
     }
-  } else {
-    // Find which panels all the selected elements have in common
-    selectionPanels = elements
-      .map(
-        ({ type }) => elementTypes.find((elType) => elType.type === type).panels
-      )
-      .reduce((commonPanels, panels) => intersect(commonPanels, panels), ALL)
-      .map((type) => {
-        switch (type) {
-          case BACKGROUND:
-            // @todo Would be good to have a general logic for panels supporting multi-selection instead.
-            // Only display when one element selected.
-            if (1 === elements.length) {
-              return { type, Panel: BackgroundPanel };
-            }
-            return null;
-          case POSITION:
-            return { type, Panel: PositionPanel };
-          case SCALE:
-            return { type, Panel: ScalePanel };
-          case ROTATION_ANGLE:
-            return { type, Panel: RotationPanel };
-          case SIZE:
-            return { type, Panel: SizePanel };
-          case FILL:
-            return { type, Panel: FillPanel };
-          case BACKGROUND_COLOR:
-            return { type, Panel: BackgroundColorPanel };
-          case COLOR:
-            return { type, Panel: ColorPanel };
-          case FONT:
-            return { type, Panel: FontPanel };
-          case STYLE:
-            return { type, Panel: StylePanel };
-          case TEXT:
-            return { type, Panel: TextPanel };
-          case VIDEO_POSTER:
-            return { type, Panel: VideoPosterPanel };
-          case MASK:
-            return { type, Panel: MaskPanel };
-          default:
-            throw new Error(`Unknown panel: ${type}`);
-        }
-      })
-      .filter((panel) => panel);
+    return panels;
   }
 
-  return [...sharedPanels, ...selectionPanels];
+  // Find which panels all the selected elements have in common
+  return elements
+    .map(
+      ({ type }) => elementTypes.find((elType) => elType.type === type).panels
+    )
+    .reduce((commonPanels, panels) => intersect(commonPanels, panels), ALL)
+    .map((type) => {
+      switch (type) {
+        case BACKGROUND:
```

Comment:
Given the code for `isBackground` above, when would this code be reached?

Reply:
That confused me too for a while, but `BACKGROUND` here just means _"include the panel allowing this element to become a background panel"_, which is visible for any (single) image or video.

This setting might be incorporated into another panel in an upcoming PR, but for now, we need it.

Reply:
Got it. Just a bad naming then. It's more of a `MAKE_BACKGROUND`.